bindings/go/Makefile.am

```diff
@@ -5,20 +5,33 @@
 
 if HAVE_GO
 
-GOPATH=$(abs_top_builddir)/bindings/go
+GOPATH=$(abs_top_builddir)/bindings/go/$(objdir)
 CGOCFLAGS=-I$(abs_top_builddir)/src -I$(top_srcdir)/src
 CGOLDFLAGS=-L$(abs_top_builddir)/src/ucp/$(objdir) -lucp -L$(abs_top_builddir)/src/ucs/$(objdir) -lucs
 UCX_SOPATH=$(abs_top_builddir)/src/ucp/$(objdir):$(abs_top_builddir)/src/ucs/$(objdir):$(abs_top_builddir)/src/ucm/$(objdir):$(abs_top_builddir)/src/uct/$(objdir)
+GOTMPDIR=$(GOPATH)/tmp
 
+export GOTMPDIR
 export GOPATH
 export CGO_CFLAGS=$(CGOCFLAGS)
 export CGO_LDFLAGS=$(CGOLDFLAGS)
 
+$(GOTMPDIR):
+	$(AM_V_at)mkdir -p $(GOTMPDIR)
+
+BUILT_SOURCES = $(GOTMPDIR)
```

Comment:
are you sure it works? because 'build' 'test' and 'bench' are not automaker targets
maybe need explicit dependency, e.g `build: $(GOTMPDIR)`
BTW, should 'test' and 'bench' depend on 'build'?

Reply:
`go test` does build internally, no need to do it explicitly in Makefile.